lib/engine-universal/src/artifact.rs

```diff
@@ -46,11 +44,9 @@ pub struct UniversalArtifact {
 }
 
 impl UniversalArtifact {
-    const MAGIC_HEADER: &'static [u8; 22] = b"\0wasmer-universal\0\0\0\0\0";
-
```

Comment:
I'd like to preserve this. I see one thing as the metadata header, and the other is the file header.
Meaning: I think it will be useful to separate the file format from the metadata serialized version.

Reply:
The metadata header itself already has a magic number and a length, so I felt that this was somewhat redundant.

Reply:
Also in the case of the universal engine, there is *only* metadata and nothing else.

Reply:
What I mean is that the magic is only relevant for the file format (thus, for the universal engine).
We don't need magic in the metadata itself, since the metadata field name in the dynamic library is already the identifier.

Therefore, I think a better abstraction is to put the magic in the engine artifact and the rest (metadata version, length, ...) on the metadata as the PR is doing

Reply:
The metadata header needs to be 16 bytes since the serialized metadata is required to be 16-byte aligned. Since the length only uses 8 bytes, we have 8 bytes left that we might as well use for a magic.

Replacing the custom header of wasmer-universal with the standard `MetadataHeader` was done to make the code more maintainable by centralizing header-related code.